client/src/App.js

```diff
@@ -69,20 +81,37 @@ class App extends Component {
     };
 
     changeLang = (lang) => {
-        const uiContext = { ...this.state.uiContext };
+        const { uiContext } = this.state;
+        const { dataContext } = this.state;
+        const { userName } = dataContext;
+        const { firstName, lastName } = userName;
+
         uiContext.lang = lang;
         uiContext.dictionary = dictionaryList[lang];
-        this.setState({ uiContext });
+
+        dataContext.userName = {
+            ...userName,
+            fullName: this.getFullName(firstName, lastName),
+            nameOrder: this.getNameOrder()
+        };
+
+        this.setState({ uiContext, dataContext });
     };
 
     setUsername = (firstName, lastName) => {
-        const dataContext = { ...this.state.dataContext };
-        dataContext.userName = `${firstName} ${lastName}`;
+        const { dataContext } = this.state;
+        dataContext.userName = {
+            firstName: firstName,
+            lastName: lastName,
```

Comment:
We can utilize JS syntax for Object initialization, and use only the property names, without values.
firstName: firstName,               --> firstName,
lastName: lastName,                --> lastName,

Reply:
Also a very good catch, thank you!